### PR TITLE
IOS-1518 Sets template selection flow

### PR DESCRIPTION
**Pull request by bududomasidet**
https://github.com/anyproto/anytype-swift/assets/15055998/11a4a5bb-5bf1-40bf-bb74-c63e4e558412

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Buttons/StandardButton.swift [hunk 1]
```diff
@@ -4,37 +4,45 @@ import AnytypeCore
...
     let action: StandardButtonAction
+    let corners: UIRectCorner
     let holdPressState: Bool // Use only for demo
     
     @State private var isPressed: Bool = false
     @Environment(\.isEnabled) private var isEnable
     
     init(
-        _ text: String,
+        _ content: Content,
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/Buttons/StandardButton.swift:24]: You can create a two convenience init with text and image. Then don't needs to refactoring everywhere

Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/Buttons/StandardButton.swift:24]: You can create a two convenience init with text and image. Then don't needs to refactoring everywhere

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/Objects/TemplatePreview.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                default:
+                    Spacer.fixedHeight(28)
+                }
+            } else {
+                Spacer.fixedHeight(28)
+            }
+        }
+    }
+    
+    var shimmer: some View {
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Templates/Views/Selection/Objects/TemplatePreview.swift:67]: Why you don't use `.redacted(.placeholder)` for shimmering state?

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                        style: .calloutRegular,
+                        color: .Button.active
+                    )
+                }
+                Spacer()
+                Button {
+                    
+                } label: {
+                    Image(asset: .X32.plus)
+                        .tint(.Button.active)
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift:36]: Tint working correctly? I tried and it didn't work for me.

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                } label: {
+                    AnytypeText(
+                        model.isEditingState ? Loc.done : Loc.edit,
+                        style: .calloutRegular,
+                        color: .Button.active
+                    )
+                }
+                Spacer()
+                Button {
+
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift:33]: Empty?

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/EditableView.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        Button {
+            onEditingTap()
+        } label: {
+            dotImage
+        }
+    }
+    
+    var dotImage: some View {
+        ZStack {
+            VisualEffectView(effect: UIBlurEffect(style: .light))
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Templates/Views/Selection/EditableView.swift:30]: Doesn't possible to use `.background(.thinMaterial)` here?

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift (hunk 4 above)
Comment (bududomasidet) [reply Anytype/Sources/PresentationLayer/Templates/Views/Selection/TemplatesSelectionView.swift:33]: Still in progress

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/EditableView.swift (hunk 5 above)
Comment (bududomasidet) [reply Anytype/Sources/PresentationLayer/Templates/Views/Selection/EditableView.swift:30]: Not the same effect unfortunately. I tried it.

File: Anytype/Sources/PresentationLayer/Templates/Assembly/TemplateModulesAssembly.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import Services
+import SwiftUI
+
+protocol TemplateModulesAssemblyProtocol {
+    func buildTemplateSelection(
```
Comment (sachnik) [inline Anytype/Sources/PresentationLayer/Templates/Assembly/TemplateModulesAssembly.swift:5]: add `@MainActor`?

File: Anytype/Sources/PresentationLayer/Templates/TemplateSelectionCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import UIKit
+import Services
+import SwiftUI
+
+protocol TemplateSelectionCoordinatorProtocol: AnyObject {
+    func showTemplatesSelection(
```
Comment (sachnik) [inline Anytype/Sources/PresentationLayer/Templates/TemplateSelectionCoordinator.swift:6]: add `@MainActor`?

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/Dataview/DataViewTemplateSelectionInteractorProvider.swift [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+import AnytypeCore
+
+
+protocol TemplateSelectionInteractorProvider {
+    func setDefaultTemplate(model: TemplatePreviewModel) async throws
+    
+    var userTemplates: AnyPublisher<[TemplatePreviewModel], Never> { get }
+}
+
+class DataviewTemplateSelectionInteractorProvider: TemplateSelectionInteractorProvider {
```
Comment (sachnik) [inline Anytype/Sources/PresentationLayer/Templates/Views/Selection/Dataview/DataViewTemplateSelectionInteractorProvider.swift:12]: if I'm not mistaken, we can mark it as `final`

File: Modules/Services/Sources/Services/Templates/File.swift [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import ProtobufMessages
```
Comment (sachnik) [inline Modules/Services/Sources/Services/Templates/File.swift:1]: looks like some redundant spaces are here in this file)

File: Anytype/Sources/PresentationLayer/Templates/Views/Selection/Objects/TemplatePreview.swift (hunk 2 above)
Comment (bududomasidet) [reply Anytype/Sources/PresentationLayer/Templates/Views/Selection/Objects/TemplatePreview.swift:67]: It is a custom shimmer. It should not shimmer some view's state, just view some placeholders without content 
Screenshot 2023-07-19 at 17 28 42

**Review by sachnik — APPROVED** [review]

**Pull request merged**